src/BootstrapMixin.js

```diff
@@ -1,11 +1,16 @@
 import React from 'react';
 import constants from './constants';
+import CustomPropTypes from './utils/CustomPropTypes';
 
 const BootstrapMixin = {
   propTypes: {
-    bsClass: React.PropTypes.oneOf(Object.keys(constants.CLASSES)),
-    bsStyle: React.PropTypes.oneOf(Object.keys(constants.STYLES)),
-    bsSize: React.PropTypes.oneOf(Object.keys(constants.SIZES))
+    bsClass: CustomPropTypes.keyOf(constants.CLASSES),
+    bsStyle: CustomPropTypes.keyOf(constants.STYLES),
+    bsSize: CustomPropTypes.keyOf(constants.SIZES)
+  },
+
+  addStyle(name) {
+    constants.STYLES[name] = name;
```

Comment:
I'm thinking we should rename `constants` to `styleMaps`? (I'm open to suggestions)

Then move this method there. It feels odd to configure this through the `BootstrapMixin`.